### PR TITLE
ECI-44 parent images for imageless children

### PR DESCRIPTION
**Pull request by iantance**
Addresses: https://dripcom.atlassian.net/browse/ECI-44

First Magento PR. There are some things in here I'm not sure about, but I thought it might be easiest to just put the PR together and have questions answered in the feedback process.

If I understand the ticket, the goal is to confirm that when a configured product has children that are not visible, we send product data that uses the parent's url and image. It appears to me that we do that, so I added a test to confirm.

### CODE REVIEW

File: devtools_m1/cypress/integration/common/admin.js [hunk 1]
```diff
@@ -111,6 +113,7 @@ Given('I have configured a configurable widget for website {string}', function(s
     "description": "This is really a widget. There are many like it, but this one is mine.",
     "shortDescription": "This is really a widget.",
     "typeId": "configurable",
+    "typeId": "configurable",
```
Comment (will-in-wi) [inline devtools_m1/cypress/integration/common/admin.js]: Is this a duplicate of the prior line?

File: devtools_m1/cypress/integration/common/admin.js [hunk 2]
```diff
@@ -119,14 +122,43 @@ Given('I have configured a configurable widget for website {string}', function(s
...
+
+Given('I have configured a configurable widget with an invisible child', function() {
+  cy.createProduct({
+    "sku": "widg-1",
+    "name": "Widget 1",
+    "description": "This is really a widget. There are many like it, but this one is mine.",
+    "shortDescription": "This is really a widget.",
+    "image": "parent_image.png",
+    "typeId": "configurable",
+    "child_visibility": 4, // not visible in either seach or catalog
```
Comment (will-in-wi) [inline devtools_m1/cypress/integration/common/admin.js]: Another approach that wouldn't require any code changes in the product builder would be to add `"visibility":4` to each child description below.

Comment (iantance) [reply devtools_m1/cypress/integration/common/admin.js]: I thought that would get overridden here: https://github.com/DripEmail/magento-m1-extension/blob/master/devtools_m1/php_utils/create_product.php#L71. Is that not the case?

Comment (will-in-wi) [reply devtools_m1/cypress/integration/common/admin.js]: Heh, good point. I think that the fix for that is to not special-case visibility:

```diff
diff --git a/devtools_m1/php_utils/create_product.php b/devtools_m1/php_utils/create_product.php
index 37738d7..b5a2eb6 100644
--- a/devtools_m1/php_utils/create_product.php
+++ b/devtools_m1/php_utils/create_product.php
@@ -59,6 +59,7 @@ class Mage_Shell_Drip_CreateProduct extends Mage_Shell_Abstract
                 "is_in_stock" => 1,
                 "qty" => 999
             ),
+            "visibility" => Mage_Catalog_Model_Product_Visibility::VISIBILITY_BOTH, //catalog and search visibility
         );
         $fullData = array_replace_recursive($defaults, $data);
 
@@ -68,8 +69,6 @@ class Mage_Shell_Drip_CreateProduct extends Mage_Shell_Abstract
             $product->$methodName($value);
         }
 
-        $product->setVisibility(Mage_Catalog_Model_Product_Visibility::VISIBILITY_BOTH); //catalog and search visibility
-
         return $product;
     }
 

```

Comment (iantance) [reply devtools_m1/cypress/integration/common/admin.js]: Hmm. Before you made that comment, I wondered if I was missing something and tried setting visibility as you suggested. Tests passed. So maybe that doesn't override after all?

File: app/code/community/Drip/Connect/Helper/Quote.php [hunk 3]
```diff
@@ -132,8 +132,20 @@ protected function prepareQuoteItemsData(Mage_Sales_Model_Quote $quote)
...
             $productVariantItem = $item;
+            $productVariantProduct = $product;
             if ($item->getProductType() === 'configurable' && $childItems[$item->getId()]) {
                 $productVariantItem = $childItems[$item->getId()];
+                $productVariantProduct = Mage::getModel('catalog/product')->load($productVariantItem->getProduct()->getId());
+            }
+
+            $productImage = $productVariantProduct->getImage();
+
+            if (empty($productImage) || $productVariantProduct->getVisibility() == Mage_Catalog_Model_Product_Visibility::VISIBILITY_NOT_VISIBLE) {
```
Comment (iantance) [inline app/code/community/Drip/Connect/Helper/Quote.php]: @will-in-wi I got all tests passing. A little investigation revealed that `getVisibility()` is returning a string (`"1"`) while the constant is an integer, so the `===` comparison was failing. I swapped it for `==`, but please let me know if there's a better way to coerce and compare here.

Comment (will-in-wi) [reply app/code/community/Drip/Connect/Helper/Quote.php]: Ah… That makes sense. I don't know if it's better, but the alternate would be:
```suggestion
            if (empty($productImage) || (int)$productVariantProduct->getVisibility() === Mage_Catalog_Model_Product_Visibility::VISIBILITY_NOT_VISIBLE) {
```

But I'd probably do what you are doing.

**Review by will-in-wi — APPROVED** [review]

**Pull request merged**